powerstore/resource_snapshotrule.go

```diff
@@ -481,12 +512,12 @@ func (r resourceSnapshotRule) Delete(ctx context.Context, req tfsdk.DeleteResour
 }
 
 // ImportState import state for existing infrastructure
-func (r resourceSnapshotRule) ImportState(ctx context.Context, req tfsdk.ImportResourceStateRequest, resp *tfsdk.ImportResourceStateResponse) {
+func (r *resourceSnapshotRule) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
```

Comment:
Please check if ImportStatePassthroughID() can be used. Applicable for all such instances